### ISSUE TITLE
Double forfeit in link causes issues, and exchanged pokemons are corrupted

### ISSUE THREAD

**Issue by Cl0v1s** [Rangi42/polishedcrystal#693]
Exchanging pokemons always result on adding a corrupted pkmn to both players parties.

![image](https://user-images.githubusercontent.com/917518/172061550-ef80937a-ceee-4478-bb72-232470562343.png)

The code responsible of that seems to be around the `.done_animation` label, from engine/link.asm
![image](https://user-images.githubusercontent.com/917518/172061589-d227641f-e378-43df-adfa-113db6115347.png)

Restoring the original Pokecrystal code fix that behavior. I have reasons to believe that the new CopyBetweenPartyAndTemp is causing that. 

Tested with BGB link emulation.

(Leaving a fight in VS mode results in a weird behavior too, maybe it's linked ?)

**Comment by vulcandth** [comment]
Master or 9-bit?

**Comment by Cl0v1s** [comment]
On master ! :)

**Comment by vulcandth** [comment]
I believe this got fixed in 9-bit. Have you tested it on 9-bit?

**Comment by Cl0v1s** [comment]
Nope ! I'll try as soon as I can :)
It's shipped on the nightly build tho.

**Comment by Cl0v1s** [comment]
It seems that it's fixed on 9bit: 
https://github.com/Rangi42/polishedcrystal/commit/a78a59b29b15f66c03418b5160ba044cf68a5df4

**Comment by FredrIQ** [comment]
>There is also an issue with fleeing from battle

Can you elaborate? This is the first time I hear of it.

**Comment by vulcandth** [comment]
I believe they are referring to Fleeing from the link battles.. I remember encountering that problem; but I don't remember if that was master on 9-bit. IIRC, if one of the player's chose to leave the battle.. they would leave normally.. but the other player would be stuck in battle.

@Cl0v1s Can you confirm? Or did you have a different issue.

**Comment by Cl0v1s** [comment]
Sorry if I was unclear.

Indeed I'm speaking about fleeing from Link Battle.

Sometimes (it happens really often but I achieved to flee safely a few times, I believe it's related to master/slave link status), when player 1 flees the fight, he can leave it without problem.

However player 2 will stay in fight and (at least in my case) a Nidoran will be sent by the "ghost" player 1.

Then the game while wait forever, since player 1 wont send any battle command because for them, the fight is over.

Indeed those problemes appear to have been resolved in 9-bit, but still not on master / releases available to the public.
